### PR TITLE
DOC: Fix website address

### LINKED ISSUE #742: Update links from fairlearn.github.io to fairlearn.org
#### Describe the issue linked to the documentation

A variety of our documentation links point to `fairlearn.github.io` rather than `fairlearn.org`. There is a redirect, but it would be good to fix these

#### Suggest a potential alternative/fix

See above

### PR DESCRIPTION
**Pull request by BrandonGoding**
resolve #742

### CODE REVIEW

**Review by riedgar-ms — APPROVED** [review]
Thanks!

File: CHANGES.md [hunk 1]
```diff
@@ -42,7 +42,7 @@
     argument `upper_bound` with the same purpose for newly enabled regression
     scenarios on `ExponentiatedGradient`.
   For a comprehensive overview of available constraints refer to the new [user
-  guide on fairness constraints for reductions methods](https://fairlearn.github.io/user_guide/mitigation.html#reductions).
+  guide on fairness constraints for reductions methods](https://fairlearn.org/user_guide/mitigation.html#reductions).
```
Comment (hildeweerts) [inline CHANGES.md:45]: This link seems to be broken - I think it should be https://fairlearn.org/main/user_guide/mitigation.html#reductions like in line 57 

@riedgar-ms

File: docs/contributor_guide/developer_call_notes/2020_07_09.rst [hunk 2]
```diff
@@ -46,7 +46,7 @@ Agenda:
 - Outreach: So far the philosophy was to wait for project to stablize before
   major outreach efforts. Most of the participants felt like it's time to do
   more on this (while ensuring we
-  `speak about fairness in the right way <https://fairlearn.github.io/contributor_guide/how_to_talk_about_fairness.html>`_).
+  `speak about fairness in the right way <https://fairlearn.org/contributor_guide/how_to_talk_about_fairness.html>`_).
```
Comment (hildeweerts) [inline docs/contributor_guide/developer_call_notes/2020_07_09.rst:49]: Similar to previous comment, I think this should be https://fairlearn.org/main/contributor_guide/how_to_talk_about_fairness.html

**Review by hildeweerts — CHANGES_REQUESTED** [review]
Thank you for this @BrandonGoding ! 

I caught two links that still seem to be broken, other than that it looks good to me!

File: CHANGES.md (hunk 1 above)
Comment (romanlutz) [reply CHANGES.md:45]: Well spotted @hildeweerts  :-)  Indeed, that's because of the branch-specific URLs (where `main` reflects latest `main` branch).

**Comment by hildeweerts** [comment]
Hi @BrandonGoding, do you need any help fixing the broken links? If you could change those we can merge your PR!

File: docs/contributor_guide/developer_call_notes/2020_07_09.rst (hunk 2 above)
Comment (romanlutz) [reply docs/contributor_guide/developer_call_notes/2020_07_09.rst:49]: This file is now deleted thanks to #745

**Comment by romanlutz** [comment]
I checked with @BrandonGoding and will create a new PR for this and close this accordingly.

**Pull request closed without merging**